dart/dynamics/BodyNode.cpp

```diff
@@ -1339,6 +1339,14 @@ void BodyNode::notifyTransformUpdate()
 {
   notifyVelocityUpdate(); // Global Velocity depends on the Global Transform
 
+  // Jacobian calculations are dependent on the parent's world transform, but
+  // not on the world transform of their own BodyNode, so they must be dirtied
+  // regardless of whether the world transform of this BodyNode is already dirty
```

Comment:
> Suppose we have a serial chain and the position of i-th joint changed then only the child BodyNodes Jacobian will be dirty when all the Jacobians of descendant BodyNodes should be dirty as well. In that sense, it seems good solution to set the flags at here instead of Joint::notifyPositionUpdate().

This is a good point. It may be the case that the regression test is only passing because of the fact that we request the transform of the final BodyNode before dirtying the Jacobian. I'm going to adjust the regression test to account for this, and I bet it's going to start failing.